document.go

```diff
@@ -356,6 +356,20 @@ func (d Document) Flag(flag DocumentFlag) bool {
 	return d.data.flag.Is(flag)
 }
 
+func (d *documentData) mergeEmbedded(other documentData, pathPrefix int) {
```

Comment:
This could be replaced by a function that takes two documents and produces one, but it has to show somehow that first is changed and no longer usable (because we don't want to fully copy slices)